scripts/deploy_preview.sh

```diff
@@ -103,3 +96,32 @@ helm upgrade -i $CHARTNAME appsmith-ee/$HELMCHART -n $NAMESPACE --create-namespa
   --set applicationConfig.APPSMITH_MONGODB_URI="mongodb+srv://$DB_USERNAME:$DB_PASSWORD@$DB_URL/$DBNAME?retryWrites=true&minPoolSize=1&maxPoolSize=10&maxIdleTimeMS=900000&authSource=admin" \
   --set applicationConfig.APPSMITH_DISABLE_EMBEDDED_KEYCLOAK=\"1\" \
   --version $HELMCHART_VERSION
+
+appsmith_deployment_timeout=1
+echo "Wait for pods to be up and running"
+until [ \
+  "$(curl -s -w '%{http_code}' -o /dev/null "$DOMAINNAME/api/v1/health")" \
+  -eq 200 ]
+do
+  echo "waiting for appsmith to be ready"
+  if [[ $appsmith_deployment_timeout -eq 300 ]];then
+     echo "timed out"
+     break
+  fi
+  sleep 5
+  appsmith_deployment_timeout=$((appsmith_deployment_timeout+5))
+done
+
+### Add maildev for recreate and new deployments only
+if [[ ${RECREATE} == "true" || ! -n "${namespace_exist}" ]]
+then
+   echo "modify docker.env to add maildev"
+   pod_name=$(kubectl get pods -n $NAMESPACE -o json | jq '.items[0].metadata.name' | tr -d '"')
+   kubectl exec -it $pod_name -n $NAMESPACE -- bash -c "cat << EOF >> /appsmith-stacks/configuration/docker.env
+   APPSMITH_MAIL_FROM=hello@appsmith.com
+   APPSMITH_MAIL_ENABLED="true"
+   APPSMITH_MAIL_HOST="maildev.default.svc.cluster.local"
+   APPSMITH_MAIL_PORT="8025"
+   EOF"
```

Comment:
I don't think this will work with this indentation. We have to remove the indentation in the command string for the `EOF` heredoc to work.

```suggestion
   kubectl exec -it $pod_name -n $NAMESPACE -- bash -c 'cat << EOF >> /appsmith-stacks/configuration/docker.env
APPSMITH_MAIL_FROM=hello@appsmith.com
APPSMITH_MAIL_ENABLED="true"
APPSMITH_MAIL_HOST="maildev.default.svc.cluster.local"
APPSMITH_MAIL_PORT="8025"
EOF'
```

Also changed to single-quotes, as otherwise the double-quotes will be lost.